be/src/formats/parquet/group_reader.cpp

```diff
@@ -44,32 +45,67 @@ Status GroupReader::get_next(vectorized::ChunkPtr* chunk, size_t* row_count) {
     bool has_more_filter = !_left_conjunct_ctxs.empty();
     Status status;
 
+    vectorized::ChunkPtr active_chunk = _create_read_chunk(_active_column_indices);
     {
         SCOPED_RAW_TIMER(&_param.stats->group_chunk_read_ns);
-        // read data into _read_chunk
-        status = _read(&count);
+        // read data into active_chunk
+        _column_reader_opts.context->selection = nullptr;
+        status = _read(_active_column_indices, &count, &active_chunk);
         _param.stats->raw_rows_read += count;
         if (!status.ok() && !status.is_end_of_file()) {
             return status;
         }
     }
 
+    vectorized::Filter* chunk_filter = nullptr;
     // dict filter
     if (has_dict_filter) {
         SCOPED_RAW_TIMER(&_param.stats->expr_filter_ns);
         SCOPED_RAW_TIMER(&_param.stats->group_dict_filter_ns);
-        _dict_filter();
-        _read_chunk->check_or_die();
+        _dict_filter(&active_chunk);
+        active_chunk->check_or_die();
+        chunk_filter = &_selection;
     }
 
     // other filter that not dict
     if (has_more_filter) {
         SCOPED_RAW_TIMER(&_param.stats->expr_filter_ns);
-        RETURN_IF_ERROR(ExecNode::eval_conjuncts(_left_conjunct_ctxs, _read_chunk.get()));
-        _read_chunk->check_or_die();
+        vectorized::FilterPtr conjunct_filter;
+        RETURN_IF_ERROR(ExecNode::eval_conjuncts(_left_conjunct_ctxs, active_chunk.get(), &conjunct_filter));
+        RETURN_IF_ERROR(_merge_filter(conjunct_filter.get(), chunk_filter));
+        active_chunk->check_or_die();
     }
 
-    *row_count = _read_chunk->num_rows();
+    count = *row_count;
+    *row_count = active_chunk->num_rows();
+    if (*row_count > 0 && !_lazy_column_indices.empty()) {
+        vectorized::ChunkPtr lazy_chunk = _create_read_chunk(_lazy_column_indices);
+        SCOPED_RAW_TIMER(&_param.stats->group_chunk_read_ns);
+        // read data into lazy chunk
+        _column_reader_opts.context->selection = chunk_filter;
+        status = _read(_lazy_column_indices, &count, &lazy_chunk);
+        if (!status.ok() && !status.is_end_of_file()) {
+            return status;
+        }
+
+        size_t num_rows = lazy_chunk->num_rows();
+        lazy_chunk->filter_range(*chunk_filter, 0, num_rows);
+        if (lazy_chunk->num_rows() != *row_count) {
+            return Status::InternalError("Unmatched row count between active and lazy column chunks");
+        }
+        active_chunk->merge(std::move(*lazy_chunk));
+    }
+
+    // reorder columns as the column order of chunk is required to be invariable
```

Comment:
https://github.com/StarRocks/starrocks/pull/8597

I will reorder column in `HiveDataSource`, so you don't have to worried about it anymore

Reply:
Great！I'll removing related codes in this PR.